orttraining/orttraining/python/experimental/orttrainer.py

```diff
@@ -395,41 +397,39 @@ def _combine_torch_model_with_loss_fn(self):
         # Input can be a list or dict
         is_list_input = (match
                          or len(input_names) >= len(ordered_input_list)
-                         or not all(x in ordered_list_kes for x in input_names))
+                         or not all(x in ordered_input_list for x in input_names))
 
-        class CombineTorchModelLossFn(torch.nn.Module):
+        class WrapModel(torch.nn.Module):
             def __init__(self, model, loss_fn, input_names):
-                super(CombineTorchModelLossFn, self).__init__()
+                super(WrapModel, self).__init__()
                 self.model = model
-                self.loss_fn = loss_fn
-                self.input_names = input_names
+                self.loss_fn_ = loss_fn
+                self.input_names_ = input_names
 
             def forward(self, *inputs):
-                # '*inputs' is given by torch trace and matches the order of 'input_names'
-                # The 'model' input might differ from 'input_names'
-                if is_list_input:
-                    input, label = inputs[:-1], inputs[-1]
-                    preds = self.model(*input)
-                    return self.loss_fn(preds, label), preds
-                else:
-                    sig = signature(self.model.forward)
-                    ordered_input_list = list(sig.parameters.keys())
-
-                    input_dict = {}
-                    for key in sig.parameters.keys():
-                        if key in self.input_names:
-                            input_dict[key] = inputs[self.input_names.index(key)]
-
-                    model_out = self.model(**input_dict)
-                    if self.loss_fn is None:
-                        return model_out
-
-                    label = inputs[-1]
-                    preds = model_out
-                    return self.loss_fn(preds, label), preds
-
-        return CombineTorchModelLossFn(self._torch_model, self.loss_fn, input_names)
+                # *inputs is given by torch trace. It is in the order of input_names.
+                # model takes input in a order (which can be obtained via inspect.signature(model.forward)) different than input_names.
+                sig = signature(self.model.forward)
+                ordered_list_keys = list(sig.parameters.keys())
 
+                input_dict = {}
+                for key in sig.parameters.keys():
+                    if key in self.input_names_:
+                        input_dict[key] = inputs[self.input_names_.index(key)]
+
+                model_out = self.model(**input_dict)
```

Comment:
Will this work as expected if the input is a list? It seems like input_dict would be empty.

Reply:
torch.onnx.export always takes dict input.